Sources/Service.swift

```diff
@@ -20,38 +24,29 @@ public class Service {
 
         task = session.dataTaskWithRequest(request) { (data, response, error) in
             self.checkStatusCodeAndData(data, urlResponse: response, error: error) { (dataResult: Result<M>) in
-                self.configuration.adaptor.serialize(fromDataResult: dataResult, result: result)
+                self.configuration.adaptor.serialize(fromDataResult: dataResult) { (jsonResult: Result<M>) in
+                    switch jsonResult {
+                    case .JSON(json: let json):
+                        let model = M(json: json)
+                        result(.Model(model))
+                    default:
+                        result(.Failure(Error.General))
+                        print("💣 damn this should not happen")
+                    }
+                }
             }
         }
 
         task!.resume()
     }
 
-    /// Receives expecte result  as defined by the `adaptor` from the a `Service` and maps this to a `Result` case `case Model(M)`
-    /// Default implementation expects `adaptor` to be     case JSON(AnyObject). If this needs to be different you need to override this method.
-    /// Typ! You can subclass 'Bar' and add a default service
-    /// - parameter call : gives the details to find the entity on the server
-    /// - parameter result : `Result<M: Mappable>` closure should be called with `case Model(M)` other cases are a failure.
-    public func perform <M: Mappable> (call: Call, toModelResult result: (Result <M>) -> ()) {
-        self.perform(call, result: { (jsonResult: Result<M>) in
-            switch jsonResult {
-            case .JSON(json: let json):
-                let model = M(json: json)
-                result(.Model(model))
-            default:
-                result(.Failure(Error.General))
-                print("💣 damn this should not happen")
-            }
-        })
-    }
-
     public func cancel() {
         task?.cancel()
     }
 
     public func checkStatusCodeAndData<M: Mappable>(data: NSData?, urlResponse: NSURLResponse?, error: NSError?, result: (Result<M>) -> ()) {
         guard error == nil else {
-            let returnError = Error.ErrorNS(error)
+            let returnError = Error.Error(Error.General)
```

Comment:
❓Ik snap niet zo goed waar `Error.Error(Error.General)` vandaan komt. Vooral dan de eerste `Error`. Er komt nogal veel Error in voor. 🖖🏼